private/bufpkg/bufcheck/rules_config.go

```diff
@@ -305,11 +303,31 @@ func newRulesConfig(
 			delete(unusedPluginNameToRuleIDs, pluginName)
 		}
 	}
-	for pluginName := range unusedPluginNameToRuleIDs {
-		// If the rule had other plugin types (see the comment on UnusedPluginNameToRuleIDs),
-		// delete the plugin name from the map
-		if _, ok := pluginNameToOtherRuleTypes[pluginName]; ok {
-			delete(unusedPluginNameToRuleIDs, pluginName)
+	// We check additional check configs. If rules are set in the related check configs, then
+	// the plugin is not considered unused. We check against all rules for all rule types.
+	allRuleIDToRule, err := getIDToRuleOrCategory(allRules)
+	if err != nil {
+		return nil, err
+	}
+	allRuleIDToCategoryIDs, err := getRuleIDToCategoryIDs(allRules)
+	if err != nil {
+		return nil, err
+	}
+	allCategoryIDToRuleIDs := getCategoryIDToRuleIDs(allRuleIDToCategoryIDs)
+	for _, checkConfig := range relatedCheckConfigs {
```

Comment:
It looks like if you specify the WithRelatedCheckConfigs option, these CheckConfigs are not *additionally* checked, rather these CheckConfigs are the *only* CheckConfigs that are checked, and the original CheckConfigs (?) from allRules are not checked. Am I wrong/does this make sense? Make sure it is additional.

Reply:
Yep! On lines 294-305, we are still checking the original configurations, we are just checking the additional rules in lieu of checking whether the plugin has rules from other rules types.